reference-adapter/mgramseva-ifix-adapter/src/main/java/org/egov/ifix/repository/MgramsevaVendorRepository.java

```diff
@@ -0,0 +1,124 @@
+package org.egov.ifix.repository;
+
+import lombok.extern.slf4j.Slf4j;
+import org.egov.ifix.cache.AdapterCache;
+import org.egov.ifix.exception.GenericCustomException;
+import org.egov.ifix.exception.HttpCustomException;
+import org.egov.ifix.models.mgramseva.*;
+import org.egov.ifix.service.AuthTokenService;
+import org.egov.ifix.utils.ApplicationConfiguration;
+import org.egov.ifix.utils.RequestHeaderUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.StringUtils;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.Optional;
+
+import static org.egov.ifix.utils.EventConstants.*;
+
+@Repository
+@Slf4j
+public class MgramsevaVendorRepository {
+    @Autowired
+    RestTemplate restTemplate;
+    @Autowired
+    ApplicationConfiguration applicationConfiguration;
+    @Autowired
+    private AuthTokenService authTokenService;
+    @Autowired
+    private RequestHeaderUtil requestHeaderUtil;
+
+    @Autowired
+    private AdapterCache<String> vendorCache;
+
+
+    public String getVendorIdByTenantId(String tenantId) {
```

Comment:
Write a TODO here that "To remove vendor references if the vendor is not mandatory in the eChallan Create Request"

Reply:
As of now it is mandatory but it could be removed in future. I'll add comment in todo.